### PR TITLE
Oled theme

### PR DESCRIPTION
**Pull request by Mehdialaoui02**

### CODE REVIEW

**Comment by dennisguse** [comment]
TODO:
* rebase to main (also remove build.gradle change)
* add a description to the PR
* squash commits (hello-commit seems to be unneeded)

One problem: What happens if a user selects theme `follow-system` and OLED-friendly?
How does OpenTracks day/night behave?

**Comment by dennisguse** [comment]
PS/ you can just force push to this branch if you changed commits.
This will update this PR.

**Comment by Mehdialaoui02** [comment]
Pull request description : 
To resolve the issue #1625, we added a button "OledTheme" that changes the theme of the app.
The new theme consists of less bright colors to save the battery. We tried to respect the visual identity of the app.

**Comment by Mehdialaoui02** [comment]
> One problem: What happens if a user selects theme `follow-system` and OLED-friendly? How does OpenTracks day/night behave?

The OledFriendly theme only applies when the user chooses day mode or if he chooses 'follow-system' and the system is in day mode. For dark mode, the button is disabled and the theme is the 'theme custom' that was defined in the app (before we added the new theme).

The following screenshots aim to describe all the cases possible for the theme application :

day_mode_oled_no
day_mode_oled_yes
night_mode_behavior
system_dark
system_day_oled_yes

File: src/main/java/de/dennisguse/opentracks/settings/PreferencesUtils.java [hunk 1]
```diff
@@ -816,7 +821,8 @@ public static void resetCustomLayoutPreferences() {
     }
 
     public static void applyNightMode() {
-        AppCompatDelegate.setDefaultNightMode(PreferencesUtils.getDefaultNightMode());
+        AppCompatDelegate.setDefaultNightMode(PreferencesUtils.getDefaultNightMode
```
Comment (dennisguse) [inline src/main/java/de/dennisguse/opentracks/settings/PreferencesUtils.java:824]: Remove this change.

File: src/main/java/de/dennisguse/opentracks/settings/UserInterfaceSettingsFragment.java [hunk 2]
```diff
@@ -38,6 +40,26 @@ public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
...
+        int nightModeFlags = getResources().getConfiguration().uiMode & Configuration.UI_MODE_NIGHT_MASK;
+        switch (nightModeFlags) {
+            case (Configuration.UI_MODE_NIGHT_YES) :
+                OLEDFriendly.setEnabled(false);
+                break;
+            case (Configuration.UI_MODE_NIGHT_NO) :
+                OLEDFriendly.setEnabled(true);
+                break;
+        }
+        OLEDFriendly.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
```
Comment (dennisguse) [inline src/main/java/de/dennisguse/opentracks/settings/UserInterfaceSettingsFragment.java:54]: Variables should be lower case.

File: src/main/res/values-fr/strings.xml [hunk 3]
```diff
@@ -65,7 +65,7 @@ Si le dispositif GPS signale des données inexactes (par exemple, la localisatio
     <!-- Activity Type -->
     <string name="activity_type_airplane">navigation aérienne</string>
     <string name="activity_type_atv">quad</string>
-    <string name="activity_type_biking">cyclisme</string>
```
Comment (dennisguse) [inline src/main/res/values-fr/strings.xml:68]: Please remove this change; translations are handled by https://hosted.weblate.org/engage/opentracks/

File: src/main/res/values/strings.xml [hunk 4]
```diff
@@ -401,6 +403,7 @@ limitations under the License.
     <string name="settings_night_mode_option_system">System</string>
     <string name="settings_night_mode_option_day">Day</string>
     <string name="settings_night_mode_option_night">Night</string>
+    <string name="settings_night_mode_option_yellow"> Yellow </string>
```
Comment (dennisguse) [inline src/main/res/values/strings.xml:406]: Leading white space "> Yellow<"?

File: src/main/res/values/colors.xml [hunk 5]
```diff
@@ -52,4 +52,12 @@ limitations under the License.
     <color name="heart_rate_zone_textcolor_4">#FFBD04</color>
     <color name="heart_rate_zone_textcolor_3">#6ACC2B</color>
     <color name="heart_rate_zone_textcolor_2">#64D1FD</color>
+
+    <!--Additions  -->
+    <color name = "yellow">#FFFF00 </color>
+    <color name = "OFF_white">#E4FBEFF1</color>
```
Comment (dennisguse) [inline src/main/res/values/colors.xml:58]: OFF -> off (also everywhere else)

File: src/main/res/values/styles.xml [hunk 6]
```diff
@@ -93,4 +93,15 @@ limitations under the License.
         <item name="colorOnSurface">@android:color/holo_red_dark</item>
     </style>
 
+    <style name="MaterialSnackbarTextButtonThemeOled" parent="@style/Widget.Material3.Button.TextButton.Snackbar">
```
Comment (dennisguse) [inline src/main/res/values/styles.xml:96]: Why?
These colors assignments shouldn't be necessary.

Comment (dennisguse) [reply src/main/res/values/styles.xml:96]: What is the purpose of these styles?

File: src/main/res/values/themes_custom.xml [hunk 7]
```diff
@@ -14,6 +14,26 @@ See the License for the specific language governing permissions and
 limitations under the License.
 -->
 <resources xmlns:android="http://schemas.android.com/apk/res/android">
+    <style name="OledFriendly" parent="Theme.Material3.Light">
```
Comment (dennisguse) [inline src/main/res/values/themes_custom.xml:17]: Why?

File: src/main/res/xml/settings_user_interface.xml [hunk 8]
```diff
@@ -18,6 +18,12 @@
         android:summary="@string/settings_ui_dynamic_colors_summary"
         android:title="@string/settings_ui_dynamic_colors_title" />
 
+    <SwitchPreferenceCompat
+        android:defaultValue="@bool/settings_ui_OLEDFriendly_default"
```
Comment (dennisguse) [inline src/main/res/xml/settings_user_interface.xml:22]: OLED should be lower case.

File: src/main/java/de/dennisguse/opentracks/AbstractActivity.java [hunk 9]
```diff
@@ -27,6 +28,7 @@
  * @author Jimmy Shih
  */
 public abstract class AbstractActivity extends AppCompatActivity {
+    public static boolean OLEDFriendly = false;
```
Comment (dennisguse) [inline src/main/java/de/dennisguse/opentracks/AbstractActivity.java:31]: I would rather move this to PreferenceUtils.

File: src/main/java/de/dennisguse/opentracks/Startup.java [hunk 10]
```diff
@@ -45,6 +45,10 @@ public void onCreate() {
         if (PreferencesUtils.shouldUseDynamicColors()) {
             DynamicColors.applyToActivitiesIfAvailable(this);
         }
+
+        if (PreferencesUtils.shouldApplyOledTheme()) {
```
Comment (dennisguse) [inline src/main/java/de/dennisguse/opentracks/Startup.java:49]: TBH: I would not use a global variable for this.
Why not check this on every `AbstractActivity.onCreate() {... PreferencesUtils.shouldApplyOledTheme() .. }`?

File: src/main/java/de/dennisguse/opentracks/AbstractActivity.java [hunk 11]
```diff
@@ -36,13 +38,31 @@ protected void onCreate(Bundle savedInstanceState) {
...
 
         setContentView(getRootView());
+        this.ThemeUpdate();
     }
 
+   public void ThemeUpdate() {
+       int nightModeFlags = getResources().getConfiguration().uiMode & Configuration.UI_MODE_NIGHT_MASK;
+       if (OLEDFriendly && (nightModeFlags == Configuration.UI_MODE_NIGHT_NO)) {
+           this.setTheme(R.style.OledFriendly);
+       }
```
Comment (dennisguse) [inline src/main/java/de/dennisguse/opentracks/AbstractActivity.java:48]: Please use autoformatting

File: src/main/java/de/dennisguse/opentracks/AbstractActivity.java [hunk 12]
```diff
@@ -36,13 +38,31 @@ protected void onCreate(Bundle savedInstanceState) {
         setVolumeControlStream(TTSManager.AUDIO_STREAM);
 
         setContentView(getRootView());
+        this.ThemeUpdate();
     }
 
+   public void ThemeUpdate() {
+       int nightModeFlags = getResources().getConfiguration().uiMode & Configuration.UI_MODE_NIGHT_MASK;
```
Comment (dennisguse) [inline src/main/java/de/dennisguse/opentracks/AbstractActivity.java:45]: This logic can be moved into PreferenceUtils (see comment regarding global variable: OLEDFriendly).

File: src/main/java/de/dennisguse/opentracks/AbstractActivity.java [hunk 13]
```diff
@@ -36,13 +38,31 @@ protected void onCreate(Bundle savedInstanceState) {
         setVolumeControlStream(TTSManager.AUDIO_STREAM);
 
         setContentView(getRootView());
+        this.ThemeUpdate();
     }
 
+   public void ThemeUpdate() {
```
Comment (dennisguse) [inline src/main/java/de/dennisguse/opentracks/AbstractActivity.java:44]: Method name should be lower case and it should be stating what it is doing like `setThemeOled()`.

File: src/main/res/values/strings.xml [hunk 14]
```diff
@@ -376,6 +376,8 @@ limitations under the License.
     <string name="settings_recording_title">Recording</string>
     <string name="settings_ui_dynamic_colors_title">Dynamic Colors</string>
     <string name="settings_ui_dynamic_colors_summary">Use user-defined Dynamic Colors (Android 12+ only; requires a manual restart)</string>
+    <string name="settings_ui_OLEDFriendly_title"> OLED Friendly</string>
```
Comment (dennisguse) [inline src/main/res/values/strings.xml:379]: Remove whitespace: `> Theme`

File: src/main/res/values/themes_custom.xml (hunk 7 above)
Comment (dennisguse) [reply src/main/res/values/themes_custom.xml:17]: And why is this not a dark theme? aka Theme.Material3.Dark

File: src/main/res/values/themes_custom.xml [hunk 15]
```diff
@@ -14,6 +14,26 @@ See the License for the specific language governing permissions and
...
+        <item name="snackbarButtonStyle">@style/MaterialSnackbarTextButtonTheme</item>
+        <item name="snackbarTextViewStyle">@style/MaterialSnackbarTextViewTheme</item>
+
+        <item name="bottomAppBarStyle">@style/Widget.OpenTracks.BottomAppBar</item>
+        <item name="floatingActionButtonStyle">@style/Widget.OpenTracks.FloatingActionButton</item>
+
+        <item name="windowActionModeOverlay">true</item>
+    </style>
 
     <style name="ThemeCustom" parent="Theme.Material3.DayNight.NoActionBar">
```
Comment (dennisguse) [inline src/main/res/values/themes_custom.xml:38]: I would rename this to ThemeOpenTracksDayNight

File: src/main/res/values/themes_custom.xml (hunk 7 above)
Comment (dennisguse) [inline src/main/res/values/themes_custom.xml:17]: Renaming this to ThemeOpenTracksOled

**Review by dennisguse — COMMENTED** [review]
I did not test it (yet).

Can you go trough my comments?

**Comment by dennisguse** [comment]
Will be implemented in #1856.

**Pull request closed without merging**